lightkurve/targetpixelfile.py

```diff
@@ -190,6 +193,87 @@ def from_archive(target, cadence='long', quarter=None, month=None,
     def __repr__(self):
         return('KeplerTargetPixelFile Object (ID: {})'.format(self.keplerid))
 
+
+    def get_sources(self, catalog=None, magnitude_limit=18, dist_tolerance=3):
+        """
+        Returns a table of stars that are centered on the target
+        of the tpf file. Current catalog supported are KIC, EPIC and Gaia DR2.
+
+        Parameters
+        -----------
+        catalog: 'KIC', 'EPIC' or 'Gaia'
+            Catalog to query. For Kepler target pixel files the default is 'KIC',
+            for K2 target pixel files the default is 'EPIC'.
+        magnitude_limit : int or float
+            Limit the returned magnitudes to only stars brighter than magnitude_limit.
+            Default 18 for both Kepmag and Gmag.
+        dist_tolerance: float
+            Maximum distance (in arcseconds) a source may be seperated from the edge
+            of the target pixel file.
+
+        Returns
+        -------
+        result : astropy.table
+            Astropy table with the following columns
+        ID : astropy.table.column (KIC & EPIC)
+            Catalog ID from catalog.
+        RAJ200: astropy.table.column (KIC & EPIC)
+            Right ascension [degrees]
+        DEJ2000: astropy.table.column (KIC & EPIC)
+            Declination [deg]
+        pmRA: astropy.table.column (KIC & EPIC)
+            Proper motion for right ascension [mas/year]
+        pmDEC: astropy.table.column (KIC & EPIC)
+        Kpmag: astropy.table.column (KIC & EPIC)
+            Magnitude in Kepler band [mag]
+        """
+
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+            else:
+                raise ValueError('Please provide a catalog.')
+
+        #Skycoord the centre of target
+        cent = SkyCoord(ra=self.ra, dec=self.dec, frame='icrs', unit=(u.deg, u.deg))
+
+        #Find the size of the TPF
+        radius = (np.max(self.flux.shape[1:2]) * 4) / 60.0
+
+        # query around centre with radius
+        data = query_catalog(cent, radius=radius, catalog=catalog)
+
+        # Load ra & dec of all tpf pixels
+        pixels_ra, pixels_dec = self.get_coordinates(cadence='all')
+
+        # Reduce calculation for astroy separation
+        pixel_radec = np.asarray([pixels_ra.ravel(), pixels_dec.ravel()])
```

Comment:
Geert has pointed out that this should only include pixels where the value is not nan. TPFs are always rectangular but have some missing nan values when the data is actually a slightly different shape.

Reply:
@christinahedges do you think this is okay? Essentially what I'm doing here is that I'm finding for a given cadence where the flux is nan -- and applying that mask to the tpf.get_coordinates... That way it wouldn't look for the separation where the flux values are nan.

But I'm wondering if nans arise for different cadence numbers, this won't be an optimal solution.